service/history/replicationTaskFetcher.go

```diff
@@ -58,12 +59,15 @@ type (
 		currentCluster string
 		sourceCluster  string
 		config         *configs.Config
+		numWorker      int
 		logger         log.Logger
 		remotePeer     admin.Client
 		rateLimiter    *quotas.DynamicRateLimiter
 		requestChan    chan *replicationTaskRequest
-		requestByShard map[int32]*replicationTaskRequest
 		shutdownChan   chan struct{}
+
+		sync.Mutex
+		workerToRequestByShard map[int]map[int32]*replicationTaskRequest
```

Comment:
Looks like worker map is very local to each worker spun up for task fetcher.  I think we should refactor the logic to have an abstraction for each of the worker as this would eliminate the need for this lock.

Reply:
> map is very local to each worker spun up for task fetcher. I think we should refactor the logic to have an abstraction for each of the worker as this would eliminate the need for this lock.

That was my original attempt, which brings more code handling the lower level worker lifecycle.
The fetch API call to source cluster is called at a relative low frequency, btw.